src/python/pants/backend/terraform/style.py

```diff
@@ -48,11 +48,11 @@ async def setup_terraform_style(setup_request: StyleSetupRequest) -> StyleSetup:
         for field_set in setup_request.request.field_sets
     )
 
-    source_files_snapshot = (
-        await Get(Snapshot, MergeDigests(sfs.snapshot.digest for sfs in source_files_by_field_set))
-        if setup_request.request.prior_formatter_result is None
-        else setup_request.request.prior_formatter_result
-    )
+    source_files_snapshot = getattr(setup_request.request, "prior_formatter_result", None)
```

Comment:
It would be more typesafe to use a union for the type in `terraform.StyleSetupRequest.request`, and then an instance check here.